### PR TITLE
optimization of trace lib

### PR DESCRIPTION
**Pull request by friztailo**
Unnecessary creation of vectors and angles has been removed.
A small change in the render lib.

### CODE REVIEW

File: lua/starfall/libs_sh/trace.lua [hunk 1]
```diff
@@ -66,16 +74,21 @@ end
...
+		start = start_vec,
+		endpos = endpos_vec,
 		filter = convertFilter(filter),
 		mask = mask,
 		collisiongroup = colgroup,
 		ignoreworld = ignworld,
 	}), "TraceResult")
 end
 
+local minbox_vec, maxbox_vec = Vector(0, 0, 0), Vector(0, 0, 0)
```
Comment (thegrb93) [inline lua/starfall/libs_sh/trace.lua:90]: Can you put all the locals together on one line?

File: lua/starfall/libs_sh/trace.lua [hunk 2]
```diff
@@ -24,6 +24,12 @@ SF.RegisterLibrary("trace")
 
 local structWrapper, util_TraceLine, util_TraceHull, util_IntersectRayWithOBB, util_IntersectRayWithPlane, util_Decal, util_PointContents, util_AimVector = SF.StructWrapper, util.TraceLine, util.TraceHull, util.IntersectRayWithOBB, util.IntersectRayWithPlane, util.Decal, util.PointContents, util.AimVector
 
+local vec_meta = getmetatable(Vector(0, 0, 0))
+local vec_SetUnpacked = vec_meta.SetUnpacked
+
+local ang_meta = getmetatable(Angle(0, 0, 0))
+local ang_SetUnpacked = ang_meta.SetUnpacked
```
Comment (thegrb93) [inline lua/starfall/libs_sh/trace.lua:31]: Change vec_meta and ang_meta to a name that doesn't conflict with the ones on line 40-41.

**Review by thegrb93 — CHANGES_REQUESTED** [review]

File: lua/starfall/libs_sh/trace.lua (hunk 2 above)
Comment (friztailo) [reply lua/starfall/libs_sh/trace.lua:31]: I don't think this change will affect anything. the same is done in render lib.

File: lua/starfall/libs_sh/trace.lua (hunk 1 above)
Comment (friztailo) [reply lua/starfall/libs_sh/trace.lua]: done

**Review by thegrb93 — APPROVED** [review]

**Pull request merged**